templates/java/Configuration.mustache

```diff
@@ -0,0 +1,102 @@
+{{>licenseInfo}}
+
+package {{invokerPackage}};
+
+{{>generatedAnnotation}}
+public class Configuration {
+    private static ApiClient defaultApiClient = new ApiClient();
+
+    /**
+     * Get the default API client, which would be used when creating API
+     * instances without providing an API client.
+     *
+     * @return Default API client
+     */
+    public static ApiClient getDefaultApiClient() {
+        return defaultApiClient;
+    }
+
+    /**
+     * Set the default API client, which would be used when creating API
+     * instances without providing an API client.
+     *
+     * @param apiClient API client
+     */
+    public static void setDefaultApiClient(ApiClient apiClient) {
+        defaultApiClient = apiClient;
+    }
+
+    public static ApiClient getDefaultApiClient(String cloudinaryUrl) {
+        updateClient(parseConfigUrl(cloudinaryUrl));
+        return defaultApiClient;
+    }
+
+    static private void updateClient(Map config) {
+        HttpBasicAuth basicAuth = (HttpBasicAuth) defaultApiClient.getAuthentication("basicAuth");
+        basicAuth.setUsername((String) config.get("api_key"));
+        basicAuth.setPassword((String) config.get("api_secret"));
+        buildCloudinaryUrl((String) config.get("cloud_name"));
+    }
+
+    static private void buildCloudinaryUrl(String cloudName) {
```

Comment:
and the function name is a bit confusing, it does not build the CLOUDINARY_URL, but updates basePath with the cloud name that was parsed from the CLOUDINARY_URL

Reply:
Fixed